module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -3610,6 +3610,8 @@ zfs_putpage(struct inode *ip, struct page *pp, struct writeback_control *wbc)
 		zil_commit(zfsvfs->z_log, zp->z_id);
 	}
 
+	dataset_kstats_update_write_kstats(&zfsvfs->z_kstat, pglen);
```

Comment:
You'll need to make a similar change to `zfs_putpages()` and `zfs_getpages()` to wire this up for FreeBSD.

One other thing to consider would be if we want to track mmap writes independently.  We could add additional fields to the kstat for this.

Reply:
I've added the appropriate calls to `zfs_putpages()` and `zfs_getpages()` for FreeBSD. As I don't have any experience with FreeBSD, I don't have a test environment available so I can't test the FreeBSD version, but the code seems straight-forward.